workmail-save-and-update-email/src/utils.py

```diff
@@ -0,0 +1,237 @@
+            if content_type == 'text/plain' and 'attachment' not in content_disposition:
+                transfer_encoding = part['Content-Transfer-Encoding']
+                text_charset = part.get_content_charset()
+                new_text_body = update_text_content(part, this_disclaimer_text, this_footer_text)
+                part.set_content(new_text_body, "plain", charset=text_charset, cte=transfer_encoding)
+            elif content_type == 'text/html' and 'attachment' not in content_disposition:
+                transfer_encoding = part['Content-Transfer-Encoding']
+                html_charset = part.get_content_charset()
+                new_html_body = update_html_content(part, this_disclaimer_text, this_footer_text)
+                if new_html_body is not None:
+                    part.set_content(new_html_body.encode(html_charset), "text", "html", cte=transfer_encoding)
+                    part.set_charset(html_charset)
+    else:
+        # Its a plain email with text/plain body
+        transfer_encoding = parsed_email['Content-Transfer-Encoding']
+        text_charset = parsed_email.get_content_charset()
+        new_text_body = update_text_content(parsed_email, this_disclaimer_text, this_footer_text)
+        parsed_email.set_content(new_text_body, "plain", charset=text_charset, cte=transfer_encoding)
+    return parsed_email
+
+def download_email(message_id):
+    """
+    This method downloads full email MIME content using GetRawMessageContent API and uses email.parser class
+    for parsing it into Python email.message.EmailMessage class.
+    Reference:
+        https://docs.python.org/3.7/library/email.message.html#email.message.EmailMessage
+        https://docs.python.org/3/library/email.parser.html
+    Parameters
+    ----------
+    message_id: string, required
+        message_id of the email to download
+    Returns
+    -------
+    email.message.Message
+        EmailMessage representation the downloaded email
+    """
+    response = workmail_message_flow.get_raw_message_content(messageId=message_id)
+    email_content = response['messageContent'].read()
+    email_generation_policy = policy.SMTP.clone(refold_source='none')
+    logger.info("Downloaded email from WorkMail successfully")
+    return email.message_from_bytes(email_content, policy=email_generation_policy)
+
+def update_workmail(message_id, content, key):
+    """
+    Uploads the updated message to an S3 bucket in your account and then updates it at WorkMail via
+    PutRawMessageContent API.
+    Reference: https://docs.aws.amazon.com/workmail/latest/adminguide/update-with-lambda.html
+    Parameters
+    ----------
+    message_id: string, required
+        message_id of the email to download
+    content: email.message.Message, required
+         EmailMessage representation the updated email
+    Returns
+    -------
+    None
+    """
+    bucket = os.getenv('UPDATED_EMAIL_BUCKET')
+    if not bucket:
```

Comment:
So since now we're reading configs from environment at several places, consider adding function like [get_env_vars](https://github.com/aws-samples/amazon-workmail-lambda-templates/blob/master/workmail-restricted-mailboxes-python/src/utils.py#L91) in your utils.py and then simply use it to read the environment variables.

Reply:
I found it odd that utils.py was where the environment was being read instead of app, which seems more natural to me. Wouldn't that make it harder to make utils.py into a shared library down the road?

Reply:
I like your modified function, but it needs to support optional env-vars. Should the optional-ity be passed in from the caller so that an exception isn't raised, or rather have the exception caught by the caller?